### ISSUE TITLE
[XGrid] Header is broken after reordering and horizontal scrolling

### ISSUE THREAD

**Issue by mindtraveller** [mui-org/material-ui-x#2148]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

The header is broken after a column is reordered and table horizontally scrolled

## Expected Behavior 🤔

The header should be displayed correctly.

## Steps to Reproduce 🕹

Steps:

1. Scroll table to the right
2. Reorder any column
3. Scroll to the left 
3. Scroll to the right -- now the header is broken

Live example: 
![Jul-14-2021 18-07-28](https://user-images.githubusercontent.com/11229620/125646012-787f36d0-9d2a-4318-bf1d-fb0ecdc39838.gif)

You  can check it yourself in this example: https://material-ui.com/components/data-grid/columns/#column-reorder

## Your Environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
```
 System:
    OS: macOS 11.4
  Binaries:
    Node: 14.15.3 - ~/.asdf/installs/nodejs/14.15.3/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.14.9 - ~/.asdf/installs/nodejs/14.15.3/bin/npm
  Browsers:
    Chrome: 91.0.4472.114
    Edge: Not Found
    Firefox: 87.0
    Safari: 14.1.1
  npmPackages:
    @material-ui/core: 4.11.1 => 4.11.1 
    @material-ui/icons: 4.5.1 => 4.5.1 
    @material-ui/lab: 4.0.0-alpha.56 => 4.0.0-alpha.56 
    @material-ui/pickers: 3.1.2 => 3.1.2 
    @material-ui/styles: 4.11.1 => 4.11.1 
    @material-ui/system:  4.11.3 
    @material-ui/types:  5.1.0 
    @material-ui/utils:  4.11.2 
    @types/react:  17.0.0 
    react: 16.12.0 => 16.12.0 
    react-dom: 16.12.0 => 16.12.0 
    typescript: 4.2.3 => 4.2.3 

  Browser:
    Google Chrome Version 91.0.4472.114
```
</details>

**Comment by dtassone** [comment]
You can downgrade the grid to version 4.0.0-alpha.30.

This bug is present in version 4.0.0-alpha.31 onwards atm.

**Comment by m4theushw** [comment]
Duplicate of #2132

**Comment by mindtraveller** [comment]
> You can downgrade the grid to version 4.0.0-alpha.30.
> 
> This bug is present in version 4.0.0-alpha.31 onwards atm.

latest version has feature that we need, so downgrading is not an option

**Comment by m4theushw** [comment]
You can try a workaround by preventing the `.MuiDataGrid-columnsContainer` div from scrolling. This is the fix proposed in #2154 for this bug.
